File: fragmentation-core-androidx/src/main/java/me/yokeyword/fragmentation/anim/DefaultHorizontalAnimator.java
package me.yokeyword.fragmentation.anim;

import android.os.Parcel;
import android.os.Parcelable;

import me.yokeyword.fragmentation.R;

/**
 * Created by YoKeyword on 16/2/5.
 */
public class DefaultHorizontalAnimator extends FragmentAnimator implements Parcelable {

    public DefaultHorizontalAnimator() {
        mEnter = R.anim.h_fragment_enter;
        mExit = R.anim.h_fragment_exit;
        mPopEnter = R.anim.h_fragment_pop_enter;
        mPopExit = R.anim.h_fragment_pop_exit;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
    }

    private DefaultHorizontalAnimator(Parcel in) {
        super(in);
    }

    public static final Creator<DefaultHorizontalAnimator> CREATOR = new Creator<DefaultHorizontalAnimator>() {
        @Override
        public DefaultHorizontalAnimator createFromParcel(Parcel source) {
            return new DefaultHorizontalAnimator(source);
        }

        @Override
        public DefaultHorizontalAnimator[] newArray(int size) {
            return new DefaultHorizontalAnimator[size];
        }
    };
}

File: fragmentation-core-androidx/src/main/java/me/yokeyword/fragmentation/anim/DefaultHorizontalAnimator.kt
package me.yokeyword.fragmentation.anim

import android.os.Parcel
import android.os.Parcelable

import me.yokeyword.fragmentation.R

/**
 * Created by YoKeyword on 16/2/5.
 */
class DefaultHorizontalAnimator : FragmentAnimator, Parcelable {

    constructor() {
        mEnter = R.anim.h_fragment_enter
        mExit = R.anim.h_fragment_exit
        mPopEnter = R.anim.h_fragment_pop_enter
        mPopExit = R.anim.h_fragment_pop_exit
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        super.writeToParcel(dest, flags)
    }

    private constructor(`in`: Parcel) : super(`in`)

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<DefaultHorizontalAnimator> = object : Parcelable.Creator<DefaultHorizontalAnimator> {
            override fun createFromParcel(source: Parcel): DefaultHorizontalAnimator {
                return DefaultHorizontalAnimator(source)
            }

            override fun newArray(size: Int): Array<DefaultHorizontalAnimator?> {
                return arrayOfNulls(size)
            }
        }
    }
}
